File: songs/clojure/src/t03_songs.clj
(ns t03-songs (:gen-class))

(defrecord Song [title tags])
(def songs [
    (map->Song {
      :title "Cantaloupe Island"
      :tags ["60s" "jazz"] })
    (map->Song {
      :title "Let It Be"
      :tags ["60s" "rock"] })
    (map->Song {
      :title "Knockin' on Heaven's Door"
      :tags ["70s" "rock"] })
    (map->Song {
      :title "Emotion"
      :tags ["70s" "pop"] })
    (map->Song {
      :title "The River" })
  ])

(defn -main [& args]
    (doseq [song songs] (println song))
  )

File: songs/scala/T03Songs.scala
case class Song(
  title: String,
  tags: Option[List[String]]
)

object T03 extends App {
  val songs: List[Song] = List(
    Song( "Cantaloupe Island",
          Some(List("60s", "jazz")) ),
    Song( "Let It Be",
          Some(List("60s", "rock")) ),
    Song( "Knockin' on Heaven's Door",
          Some(List("70s", "rock")) ),
    Song( "Emotion",
          Some(List("70s", "pop")) ),
    Song( "The River", None )
  )

  songs map (song => println(song))
}
